Foundation/testsuite/src/ProcessRunnerTest.cpp

```diff
@@ -0,0 +1,332 @@
+#include "Poco/Util/PropertyFileConfiguration.h"
+#include "Poco/Format.h"
+#include "Poco/Path.h"
+#include "Poco/File.h"
+#include "Poco/FileStream.h"
+#include "Poco/Environment.h"
+#include <fstream>
+#include <iostream>
+
+
+//using namespace IoT::Devs::Util;
+using namespace Poco;
+using namespace Poco::Util;
+//using IoT::Devs::Util::enableLogging;
+
+
+ProcessRunnerTest::ProcessRunnerTest(const std::string& name): 
+	CppUnit::TestCase(name)
+{
+	//enableLogging("debug", "%Y-%m-%d %H:%M:%S.%i [%s]: %t");
+}
+
+
+ProcessRunnerTest::~ProcessRunnerTest()
+{
+}
+
+
+void ProcessRunnerTest::testPIDFile()
+{
+	std::string pidFile = Path::tempHome() + "test.pid";
+	std::string pidFiled = Path::tempHome() + "testd.pid";
+
+	{
+		PIDFile f;
+		assertTrue (f.getName().empty());
+		assertTrue (f.getPID() == PIDFile::INVALID_PID);
+		assertFalse (File(pidFile).exists());
+		assertFalse (File(pidFiled).exists());
+
+		f.setName(pidFile, false);
+		assertTrue (f.getName() == pidFile);
+		assertTrue (f.getPID() != PIDFile::INVALID_PID);
+		assertTrue (File(pidFile).exists());
+		assertTrue (!File(pidFiled).exists());
+
+#ifdef _DEBUG
+		f.setName(pidFile, true);
+		assertTrue (f.getName() == pidFiled);
+		assertTrue (f.getPID() != PIDFile::INVALID_PID);
+		assertTrue (!File(pidFile).exists());
+		assertTrue (File(pidFiled).exists());
+#endif
+	}
+	assertFalse (File(pidFile).exists());
+	assertFalse (File(pidFiled).exists());
+
+	{
+		PIDFile f(pidFile);
+#ifdef _DEBUG
+		std::string pf = pidFiled;
+#else
+		std::string pf = pidFile;
+#endif
+
+		assertTrue (f.getName() == pf);
+		assertTrue (File(pf).exists());
+		assertTrue (f.getPID() != PIDFile::INVALID_PID);
+
+		assertTrue (f.exists());
+	}
+	assertFalse (File(pidFile).exists());
+	assertFalse (File(pidFiled).exists());
+
+	{
+		PIDFile f(pidFile, false);
+		assertTrue (f.getName() == pidFile);
+		assertTrue (File(pidFile).exists());
+		assertTrue (f.getPID() != PIDFile::INVALID_PID);
+
+		assertTrue (f.exists());
+	}
+	assertFalse (File(pidFile).exists());
+	assertFalse (File(pidFiled).exists());
+
+	{
+		PIDFile f(pidFile, true, false);
+#ifdef _DEBUG
+		std::string pf = pidFiled;
+#else
+		std::string pf = pidFile;
+#endif
+
+		assertTrue (f.getName() == pf);
+		assertTrue (!File(pf).exists());
+		assertTrue (f.getPID() == PIDFile::INVALID_PID);
+
+		f.create(false);
+		assertTrue (f.exists());
+	}
+	assertFalse (File(pidFile).exists());
+	assertFalse (File(pidFiled).exists());
+}
+
+
+void ProcessRunnerTest::testProcessRunner()
+{
+	std::string name("TestApp");
+	std::string cmd;
+	
+#ifdef _DEBUG
+	name += 'd';
+#endif
+
+	cmd = name;
+
+	// non-auto start, no PID
+	{
+		std::vector<std::string> args;
+		ProcessRunner pr(cmd, args, "", ProcessRunner::NO_OUT, 10, false);
+		assertTrue (pr.cmdLine() == cmdLine(cmd, args));
+		assertFalse (pr.running());
+		pr.start();
+		Stopwatch sw; sw.start();
+		while (!pr.running())
+			checkTimeout(sw, "Waiting for process to start", 1000);
```

Comment:
make this `checkTimeout(sw, "Waiting for process to start", 1000, __LINE__);`,
and this:
https://github.com/pocoproject/poco/blob/95a107b5c0f7a1b2b987433b0a51db1a27eac7ca/Foundation/testsuite/src/ProcessRunnerTest.cpp#L303

into this:

```c++
void ProcessRunnerTest::checkTimeout(const Stopwatch& sw, const std::string& msg, int timeoutMS, int line)
{
	if (sw.elapsedSeconds()*1000 > timeoutMS)
	{
		throw Poco::TimeoutException(
			Poco::format("ProcessRunner::checkTimeout(): %s, line: %d", msg, line));
	}
	Thread::sleep(10);
}
```

Reply:
Fixed in https://github.com/pocoproject/poco/pull/4225/commits/a08f5b5bb749a8983dcbdde145d9d4e398caebf3